genesyscloud/architect_flow/resource_genesyscloud_architect_flow_utils.go

```diff
@@ -35,8 +41,109 @@ func GenerateFlowResource(resourceLabel, srcFile, fileContent string, forceUnloc
 	return flowResourceStr
 }
 
-// setFileContentHashToNil This operation is required after a flow update fails because we want Terraform to detect changes
-// in the file content hash and re-attempt an update, should the user re-run terraform apply without making changes to the file contents
+// architectFlowResolver downloads and processes an architect flow from Genesys Cloud.
+// It creates a subdirectory, downloads the flow file, and updates the resource configuration.
+//
+// Parameters:
+//   - flowId: The ID of the architect flow to resolve
+//   - exportDirectory: The base directory where the flow will be exported
+//   - subDirectory: The subdirectory name to create within the export directory
+//   - configMap: Configuration map containing resource settings
+//   - meta: Provider metadata containing client configuration
+//   - resource: Resource information for the architect flow
+//
+// Returns:
+//   - error: Returns an error if any operation fails, nil otherwise
+func architectFlowResolver(flowId, exportDirectory, subDirectory string, configMap map[string]any, meta any, resource resourceExporter.ResourceInfo) (err error) {
+	defer func() {
+		if err != nil {
+			err = fmt.Errorf("caught in architectFlowResolver: %w", err)
+		}
+	}()
+
+	var (
+		sdkConfig = meta.(*provider.ProviderMeta).ClientConfig
+		proxy     = newArchitectFlowProxy(sdkConfig)
+		ctx       = context.Background()
+		flowName  string
+	)
+
+	flow, resp, err := proxy.GetFlow(ctx, flowId)
+	if err != nil {
+		log.Printf("Failed to establish flow name. Error: %s. API Response: %s", err.Error(), resp)
+	} else if flow != nil && flow.Name != nil {
+		flowName = sanitizeFlowName(*flow.Name)
+	}
+
+	downloadUrl, err := proxy.generateDownloadUrl(flowId)
+	if err != nil {
+		return err
+	}
+
+	log.Printf("Creating subfolder '%s' inside '%s'", subDirectory, exportDirectory)
+	fullPath := filepath.Join(exportDirectory, subDirectory)
+	if err := os.MkdirAll(fullPath, os.ModePerm); err != nil {
+		return err
+	}
+	log.Printf("Successfully created subfolder '%s' inside '%s'", subDirectory, exportDirectory)
+
+	filename := fmt.Sprintf("%s-%s.yaml", flowName, flowId)
+	log.Printf("Downloading export flow '%s' to '%s' from download URL", flowId, fullPath)
+	if resp, err := files.DownloadExportFile(fullPath, filename, downloadUrl); err != nil {
+		log.Printf("Failed to download flow file: %s", err.Error())
+		if resp != nil {
+			log.Printf("API Response: " + resp.String())
+		}
+		return err
+	}
+	log.Printf("Successfully downloaded export flow '%s' to '%s'", flowId, fullPath)
+
+	log.Printf("Updating resource config and state file for flow '%s'", flowId)
+	updateResourceConfigAndState(configMap, resource, exportDirectory, subDirectory, filename)
+	return err
+}
+
+// sanitizeFlowName will replace all forward slashes, backslashes and white spaces with an underscore
+func sanitizeFlowName(s string) string {
```

Comment:
This is to ensure a valid filename is generated right? Should we sanitize all invalid filename characters then? https://stackoverflow.com/a/31976060
Should we pull this into a helper utility?

Reply:
The intent was to make sure no directory traversal stuff can be done. Also escaped whitespaces in the CLI are an eye sore, so I removed them also. Otherwise, I don't think there is any risk since the `.yaml` extension is hard-coded at the end